tests/Instances.hs

```diff
@@ -122,6 +122,8 @@ instance Arbitrary RectC where
 instance Arbitrary Rectangle where
   arbitrary = Rectangle <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
 
+instance Arbitrary RationalRect where
+  arbitrary = RationalRect <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
```

Comment:
This seems general enough to be in the Instances module (frankly, I'm surprised no one else has needed it up to this point) and so it is there :)

Reply:
Does that somehow limit the values it can take, or are we also testing out-of-range `RationalRect`s? Because I'm pretty sure the type doesn't and can't enforce the range limits.

Reply:
We're also testing out ot range arguments here.  Since the type doesn't enforce these limits, I've seen users not care much about them either (stretching a floating window over three monitors is easily done with the appropriate `RationalRect`, for example). 

It would be an easy fix to only accept value between 0 and 1 here; not sure what's best

Reply:
Okay I decided to restrict `RationalRects` to be at least between 0 and 1.  There is no coherence between these values, so this still leaves room for windows to go outside of the rectangle.  However, that's something that can (and will) happen quite regularly so I don't think it's that much of an issue